### PR TITLE
Setup improvement

### LINKED ISSUE #31: Better management of dependencies
human_aware_rl is not listed on pip right now.

- Remove conda from git workflow
- Clean up dependencies stuff in the repo too
- Updating documentation in README

### PR DESCRIPTION
**Pull request by jyan1999**
Removed install.sh
Added an __init__.py file for human_aware_rl directory so it can be properly recognized as a module
Updated workflow file to simplify setup and added pip cache action
Included a requirements.txt file for caching github workflow, installation can now be done via pip install -r requirements.txt
Updated test suite to allow unittest discovery (though it is pretty limited. More comprehensive discovery requires nontrivial refactoring)
Included temporary fix to increase tests stability by setting sgd-size = training-size (it doesn't completely solve the issue, but does significantly increase stability)
Updated the README to reflect the changes and fixed some errors

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #31 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works --- existing tests should be sufficient

### CODE REVIEW

File: run_tests.sh [hunk 1]
```diff
@@ -4,22 +4,43 @@ cd ./human_aware_rl
```
Comment (micahcarroll) [inline run_tests.sh:4]: Part of me feels like we should discontinue this file and have the run-tests command be the more standard auto-test discovery that unittest provides (https://stackoverflow.com/a/15630454/5578742).

Regarding profiling, I usually just use [this decorator](https://github.com/HumanCompatibleAI/overcooked_ai/blob/df2fc6ceaa19203250aa68755b861602d8ff586b/src/overcooked_ai_py/utils.py#L206) around any function I'm currently interested in profiling, or use the built-in profiler to PyCharm (which wouldn't require having `cProfile` or `gprof2dot ` in the requirements for the repo).

That being said, if you think this set up might be useful for you in the future to be able to quickly profile tests, I'm open to keeping it :)

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#this file provides an alternative to set up via install.sh
```
Comment (micahcarroll) [inline requirements.txt:1]: Might be helpful to be a bit more clear about what you mean by `alternative to set up via install.sh`. Do you mean that they are fully equivalent alternatives?

My current understanding is that `requirements.txt` does exactly the same thing as `install.sh` (except maybe installing git lfs)? If so, can't we just delete `install.sh` and use `requirements.txt` everywhere instead?

**Review by micahcarroll — COMMENTED** [review]
This looks good to me! Thanks for all the documentation changes. I've left a couple of minor comments and questions!

File: README.md [hunk 3]
```diff
@@ -215,11 +232,13 @@ ModuleNotFoundError: No module named 'human_aware_rl.data_dir'
 , please run 
 
 ```
-./run_tests.sh
+./install.sh
```
Comment (micahcarroll) [inline README.md:235]: This file doesn't exist anymore right?

Comment (jyan1999) [reply README.md:235]: Ah right yes let me fix that

**Review by micahcarroll — APPROVED** [review]
LGTM! Thanks Jackson!

**Pull request merged**